### PR TITLE
Translate MAVLink telemetry to CRSF on the handset

### PR DESCRIPTION
**Pull request by MUSTARDTIGERFPV**
## What this PR accomplishes

When using the new MAVLink link mode, we no longer have access to FC telemetry like GPS coordinates, flight mode, altitude, etc. That data *is* however available in the MAVLink stream; this PR adds functionality to the TX to interpret the MAVLink messages going through it to your GCS and convert relevant parameters into CRSF telemetry, putting values like altitude back in your handset and in your EdgeTX telemetry logs.

## What's needed to get it out of draft state

1. Testing is needed with ESP8285-based transmitters, which have less free memory to play around with.
2. It's likely that explicitly declaring buffers (and eschewing the default 4 channels' worth of buffers) would reduce memory usage by ~800k.
3. Real-world testing - my bench FC's GPS values are pretty boring - things might get funny in the real world.

## Preview

mavlink-rc-crsf

### CODE REVIEW

File: src/src/tx_main.cpp [hunk 1]
```diff
@@ -1510,6 +1512,9 @@ void loop()
         {
           // raw mavlink data - forward to USB rather than handset
           uint8_t count = CRSFinBuffer[1];
+          // Convert to CRSF telemetry where we can
+          convert_mavlink_to_crsf_telem(CRSFinBuffer, count, handset);
+          // If we have a backpack
```
Comment (wvarty) [inline src/src/tx_main.cpp:1517]: I think this comment belongs one line down. The line directly below this is the write to the USB UART, not the backpack.

**Review by wvarty — APPROVED** [review]

**Comment by dmazan** [comment]
Excellent addition.  Just what I was looking for.  I'll try it as soon as I can.

File: src/src/tx_main.cpp [hunk 2]
```diff
@@ -541,19 +543,35 @@ void ICACHE_RAM_ATTR SendRCdataToRF()
       otaPkt.std.type = PACKET_TYPE_MSPDATA;
       if (OtaIsFullRes)
       {
-        otaPkt.full.msp_ul.packageIndex = MspSender.GetCurrentPayload(
```
Comment (JyeSmith) [inline src/src/tx_main.cpp:544]: From here down looks to revert some changes in https://github.com/ExpressLRS/ExpressLRS/pull/2798

File: src/lib/OTA/OTA.h [hunk 3]
```diff
@@ -70,6 +70,13 @@ typedef struct {
                     tlmFlag:1;
             uint8_t payload[ELRS4_MSP_BYTES_PER_CALL];
         } msp_ul;
+        /** PACKET_TYPE_MAV **/
+        struct {
+            uint8_t packageIndex:5,
+                    tlmFlag:1,
+                    spare:2;
+            uint8_t payload[ELRS4_MSP_BYTES_PER_CALL];
+        } mav_ul;
```
Comment (JyeSmith) [inline src/lib/OTA/OTA.h:79]: Sad noises... https://github.com/ExpressLRS/ExpressLRS/pull/2798

**Comment by JyeSmith** [comment]
GTG after address the above.

**Comment by wvarty** [comment]
Moving this PR to https://github.com/ExpressLRS/ExpressLRS/pull/2808 after a rebase

**Pull request closed without merging**